Lombiq.NodeJs.Extensions/scripts/get-assets-config.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @summary A script to return the configuration from assets-to-copy.json or package.json (assetsToCopy).
+ * @description This script reads a configuration object from a file in the consuming project named
+ *              "assets-to-copy.json" or the "assetsToCopy" node in package.json, and expects the following format:
+ *                [
+ *                  {
+ *                    "sources": [ "Assets/Images" ],
+ *                    "target": "wwwroot/images"
+ *                  },
+ *                  {
+ *                    "sources": [
+ *                      "node_modules/marvelous/dist"
+ *                      "node_modules/wonderful/bin"
+ *                    ],
+ *                    "pattern": "*",
+ *                    "target": "wwwroot"
+ *                  }
+ *                  // more groups, if needed
+ *                ]
+ */
+const { readFile } = require('node:fs/promises');
+const path = require('path');
+
+const assetsFileName = 'assets-to-copy.json';
+const assetsKeyInPackageJson = 'assetsToCopy';
+
+function logError(message) {
+    process.stderr.write(message + '\n');
+}
+
+function isValid(assetsGroup) {
+    const errors = [];
+    if (!Array.isArray(assetsGroup.sources)) {
+        errors.push('sources must be an array of strings');
+    }
+    if (assetsGroup.pattern && typeof assetsGroup.pattern !== 'string') {
+        errors.push('pattern must be a glob string');
+    }
+    if (typeof assetsGroup.target !== 'string') {
+        errors.push('target must be a string');
+    }
+    if (errors.length > 0) {
+        logError(`Invalid asset group: ${JSON.stringify(assetsGroup)}: ${errors.join(', ')}.`);
+    }
+    return errors.length === 0;
+}
+
+async function getAssetsConfig({ directory, verbose } = { directory: process.cwd(), verbose: false }) {
+    const log = (message) => { if (verbose) process.stdout.write(message); };
+    const logn = (message) => { log(message + '\n'); };
+
+    const assetsJsonPath = path.resolve(directory, assetsFileName);
+    const packageJsonPath = path.resolve(directory, 'package.json');
+
+    return Promise.resolve()
+        .then(() => log(`Reading configuration from ${assetsJsonPath} ... `))
+        .then(() => readFile(assetsJsonPath, 'utf-8'))
```

Comment:
Why? You can have the first log outside of the promise, and start with `readFile` so you don't need the `Promise.resolve()`.

Reply:
I loved how fluent this is!

Reply:
Using a bunch of `.then()`s doesn't seem to respect the spirit of a fluent interface. That's just method chaining for aesthetic reasons.

Reply:
Or how about making the function `async` and using `await`s instead?

Reply:
Much nicer!